src/key/key.js

```diff
@@ -510,10 +510,10 @@ class Key {
       }
     }
     if (!users.length) {
-      // eslint-disable-next-line @typescript-eslint/no-throw-literal
+      // eslint-disable-next-line @typescript-eslint/only-throw-error
       throw exception || new Error('Could not find primary user');
     }
-    await Promise.all(users.map(async function (a) {
+    await Promise.all(users.map(a => {
       return a.selfCertification.revoked || a.user.isRevoked(a.selfCertification, null, date, config);
     }));
```

Comment:
I know that linters like to complain about "useless" uses of `async`, but I think it makes it clearer here to explain why the `Promise.all` is needed. Perhaps we could do:
```suggestion
    await Promise.all(users.map(async a => {
      a.selfCertification.revoked || await a.user.isRevoked(a.selfCertification, null, date, config);
    }));
```
(the code here is also a bit strange because we rely on the side-effect of setting the `revoked` property rather than just using the return value directly, but OK).

Reply:
The linter now throws if `awaits` are redundant (or missing) so I think it's probably more confusing to specify them unnecessarily

Reply:
`isRevoked` is async so that one is not unnecessary

Reply:
By marking the function as `async`, it still returns a promise.
`a.selfCertification.revoked`, by contrast, _isn't_ a promise, so the version I suggested is actually more type-consistent because now it always returns a promise, instead of only sometimes.